crates/e2e/tests/e2e/eth_flow.rs

```diff
@@ -0,0 +1,119 @@
+            partially_fillable: quote.partially_fillable,
+            quote_id: quote_response.id.expect("No quote id"),
+        })
+    }
+
+    fn to_cow_swap_order(&self, ethflow: &CoWSwapEthFlow, weth: &WETH9) -> Order {
+        // Each ethflow user order has an order that is representing
+        // it as EIP1271 order with a different owner and valid_to
+        OrderBuilder::default()
+            .with_kind(OrderKind::Sell)
+            .with_sell_token(weth.address())
+            .with_sell_amount(self.0.sell_amount)
+            .with_fee_amount(self.0.fee_amount)
+            .with_receiver(Some(self.0.receiver))
+            .with_buy_token(self.0.buy_token)
+            .with_buy_amount(self.0.buy_amount)
+            .with_valid_to(u32::MAX)
+            .with_app_data(self.0.app_data.0)
+            .with_eip1271(ethflow.address(), hex!("").into())
+            .build()
+    }
+
+    pub fn include_slippage_bps(&self, slippage: u16) -> Self {
+        let max_base_point = 10000;
+        if slippage > max_base_point {
+            panic!("Slippage must be specified in base points");
+        }
+        ExtendedEthFlowOrder(EthflowOrder {
+            buy_amount: self.0.buy_amount * (max_base_point - slippage) / max_base_point,
+            ..self.0
+        })
+    }
+
+    pub async fn status(&self, contracts: &Contracts) -> EthFlowOrderOnchainStatus {
+        contracts
+            .ethflow
+            .orders(Bytes(self.hash(contracts).await.0))
+            .call()
+            .await
+            .expect("Couldn't fetch native token balance")
+            .into()
+    }
+
+    pub async fn mine_order_creation(
+        &self,
+        owner: &Account,
+        ethflow: &CoWSwapEthFlow,
+    ) -> TransactionResult {
+        tx_value!(
+            owner,
+            self.0.sell_amount + self.0.fee_amount,
+            ethflow.create_order(self.0.encode())
+        )
+    }
+
+    async fn hash(&self, contracts: &Contracts) -> H256 {
+        let domain_separator = DomainSeparator(
+            contracts
+                .gp_settlement
+                .domain_separator()
+                .call()
+                .await
+                .expect("Couldn't query domain separator")
+                .0,
+        );
+        H256(hashed_eip712_message(
+            &domain_separator,
+            &self
+                .to_cow_swap_order(&contracts.ethflow, &contracts.weth)
+                .data
+                .hash_struct(),
+        ))
+    }
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum EthFlowOrderOnchainStatus {
+    Invalidated,
+    Created(H160, u32),
+    Free,
+}
+
+impl From<(H160, u32)> for EthFlowOrderOnchainStatus {
+    fn from((owner, valid_to): (H160, u32)) -> Self {
+        match owner {
+            owner if owner == NO_OWNER => Self::Free,
+            owner if owner == INVALIDATED_OWNER => Self::Invalidated,
+            _ => Self::Created(owner, valid_to),
+        }
+    }
+}
```

Comment:
I agree this is very much key logic of the contract, on the other hand since it hasn't been introduced so far I suppose it isn't really important in other sections of the code outside of the test. I'll be making other changes to how the eth flow work, so I might eventually move this out of the test and into the main code.

(Note that `INVALIDATED_OWNER` is part of the main refunder code, so I also put `NO_OWNER` there for consistency.)